sentinel-ai-core/src/main/java/com/phonepe/sentinelai/core/agent/Agent.java

```diff
@@ -229,6 +232,12 @@ public A registerTools(Map<String, ExecutableTool> callableTools) {
         return self;
     }
 
+    public A registerAgentMessagesPreProcessor(AgentMessagesPreProcessor agentMessagesPreProcessor) {
+        this.agentMessagesPreProcessors.add(agentMessagesPreProcessor);
+        log.info("Registering messages pre-processor: {} for agent: {}", agentMessagesPreProcessor.getClass().getSimpleName(), name());
```

Comment:
debug

Reply:
Instead of only one, add 2 methods:
- `register(List)`
- `register(PP... processors)` -> delegate to previous method